storage/shard.go

```diff
@@ -425,3 +442,17 @@ func (s *dbShard) Flush(ctx context.Context, blockStart time.Time, pm persist.Ma
 
 	return multiErr.FinalError()
 }
+
+func (s *dbShard) CleanupFileset(namespace string, earliestToRetain time.Time) error {
+	filePathPrefix := s.opts.GetCommitLogOptions().GetFilesystemOptions().GetFilePathPrefix()
+	multiErr := xerrors.NewMultiError()
+	expired, err := s.filesetBeforeFn(filePathPrefix, namespace, s.ID(), earliestToRetain)
+	if err != nil {
+		detailedErr := fmt.Errorf("encountered errors when getting fileset files for prefix %s namespace %s shard %d: %v", filePathPrefix, namespace, s.ID(), err)
+		multiErr = multiErr.Add(detailedErr)
+	}
+	if err := s.deleteFilesFn(expired); err != nil {
+		multiErr = multiErr.Add(err)
+	}
+	return multiErr.FinalError()
```

Comment:
Yeah if we ever do consolidate data, we'll need to move it up, but for now feels like it's more consistent that way since that's also how flush and bootstrap are structured.